### PR TITLE
Update RayCast class ref

### PR DESCRIPTION
**Pull request by cbscribe**
- Add descriptions to member variables
- Add examples
- Minor grammar/wording

### CODE REVIEW

File: doc/base/classes.xml [hunk 1]
```diff
@@ -40647,14 +40659,14 @@
 			<return type="Vector3">
 			</return>
 			<description>
-				Returns the normal of the intersecting object shape face containing the collision point.
+				Returns the normal of the intersecting object's shape at the collision point.
```
Comment (turtletooth) [inline doc/base/classes.xml]: I'm new to Godot, so I may be wrong, but removing the word face may take away some clarity.  A normal is generally associated with one face. If so, perhaps:
> Returns the normal of the face of the intersecting object at the collision point.

Comment (cbscribe) [reply doc/base/classes.xml]: I'm open to it.  "shape face" just struck me as off. Too many uses of "of the", though.

> Returns the intersecting object's shape's normal at the collision point.

Comment (turtletooth) [reply doc/base/classes.xml]: I understand.  I had the same struggle as I was writing mine.  Yours feels less verbose, but perhaps not as specific.

**Comment by turtletooth** [comment]
Good work @cbscribe - Thank you.

File: doc/base/classes.xml [hunk 2]
```diff
@@ -40752,11 +40752,11 @@
 		Query the closest object intersecting a ray.
 	</brief_description>
 	<description>
-		A RayCast represents a line from its origin to its destination position [code]cast_to[/code], it is used to query the 3D space in order to find the closest object intersecting with the ray.
+		A RayCast represents a line from its origin to its destination position, [code]cast_to[/code]. It is used to query the 2D space in order to find the closest object along the path of the ray.
```
Comment (Noshyaar) [inline doc/base/classes.xml]: I think blank lines between paragraphs aren't necessary?

**Comment by Noshyaar** [comment]
Please squash commits together.

**Comment by cbscribe** [comment]
Squashed commits

File: doc/base/classes.xml (hunk 2 above)
Comment (Noshyaar) [reply doc/base/classes.xml]: https://github.com/GDquest/godot-docs/blob/9cb30a854c784e1426e8ffcc3f20c1f7e3a3bfb4/community/contributing/updating_the_class_reference.rst

> Write in a clear and simple language. Always follow the :ref:`class_reference_styleguide` to keep your descriptions short and easy to read. Do not leave empty lines in the descriptions: each line in the XML file will result in a new paragraph.

**Pull request merged**